### PR TITLE
NEW Availibility dictionnary new column unit and number

### PR DESCRIPTION
**Pull request by atm-quentin**
image

### CODE REVIEW

File: htdocs/install/mysql/migration/15.0.0-16.0.0.sql [hunk 1]
```diff
@@ -276,3 +276,13 @@ UPDATE llx_c_ticket_type SET use_default=1 WHERE code='OTHER' AND NOT EXISTS(SEL
 
 UPDATE llx_c_country SET eec=0 WHERE eec IS NULL;
 ALTER TABLE llx_c_country MODIFY COLUMN eec tinyint DEFAULT 0 NOT NULL;
+
+ALTER TABLE llx_c_availability ADD COLUMN type_duration char(1);
+ALTER TABLE llx_c_availability ADD COLUMN number real DEFAULT 0;
+
+delete from llx_c_availability WHERE rowid IN (1,2,3,4,5);
```
Comment (eldy) [inline htdocs/install/mysql/migration/15.0.0-16.0.0.sql:283]: Do not include instruction that may destroy customization of user into a migration. Instead,
just keep the insert (there is a protection of duplicate due to the unique key on code) and after, for each insert, make an update to update the new field when code = the code.
Do not rely on rowid that may be different on each installation.

Comment (atm-quentin) [reply htdocs/install/mysql/migration/15.0.0-16.0.0.sql]: I made some modifications, are you ok with that ?

**Comment by atm-maxime** [comment]
I think "number" is an incorrect name for a SQL column name. Did you check if it's a reserved word ?

**Comment by atm-quentin** [comment]
i renamed it into qty

File: htdocs/install/mysql/migration/15.0.0-16.0.0.sql [hunk 2]
```diff
@@ -279,4 +279,11 @@ ALTER TABLE llx_propal ADD last_main_doc VARCHAR(255) NULL AFTER model_pdf;
 
 UPDATE llx_c_country SET eec=0 WHERE eec IS NULL;
 ALTER TABLE llx_c_country MODIFY COLUMN eec tinyint DEFAULT 0 NOT NULL;
-
+ALTER TABLE llx_c_availability ADD COLUMN type_duration char(1);
```
Comment (eldy) [inline htdocs/install/mysql/migration/15.0.0-16.0.0.sql:282]: Do not use 'char', use always varchar, even for 1 char.

File: htdocs/install/mysql/tables/llx_c_availability.sql [hunk 3]
```diff
@@ -23,6 +23,8 @@ create table llx_c_availability
 	rowid		integer	 	AUTO_INCREMENT PRIMARY KEY,
 	code		varchar(30) NOT NULL,
 	label		varchar(128) NOT NULL,
+	type_duration	char(1),
```
Comment (eldy) [inline htdocs/install/mysql/tables/llx_c_availability.sql:26]: Do not use 'char', use always varchar, even for 1 char.

File: htdocs/install/mysql/migration/15.0.0-16.0.0.sql (hunk 2 above)
Comment (atm-quentin) [reply htdocs/install/mysql/migration/15.0.0-16.0.0.sql]: i did it

File: htdocs/install/mysql/tables/llx_c_availability.sql (hunk 3 above)
Comment (atm-quentin) [reply htdocs/install/mysql/tables/llx_c_availability.sql]: i did it

**Pull request merged**